core/src/lib.rs

```diff
@@ -1,10 +1,60 @@
-#![warn(missing_docs)]
-//! The unimarkup-rs crate is the official implementation of the [Unimarkup specification](https://github.com/Unimarkup/Specification/).
-
-// TODO: set to private modules that don't have to be public
-pub mod document;
-pub mod elements;
-pub mod metadata;
-pub mod parser;
-pub mod security;
-pub mod unimarkup;
+pub use unimarkup_commons as commons;
+pub use unimarkup_parser as parser;
+pub use unimarkup_render as render;
+
+use crate::commons::config::output::OutputFormatKind;
+use crate::commons::config::Config;
+use crate::parser::document::Document;
+use crate::render::html::render::HtmlRenderer;
+use crate::render::html::Html;
+use crate::render::log_id::RenderError;
+use crate::render::render::{OutputFormat, Renderer};
+
+pub struct Unimarkup {
+    doc: Document,
+}
+
+impl Unimarkup {
+    /// Parses Unimarkup content, and returns a [`Unimarkup`] struct to render the content to supported formats.
+    ///
+    /// # Arguments
+    ///
+    /// * `um_content` - String containing Unimarkup elements.
+    /// * `config` - Unimarkup configuration to be used on top of preambles.
+    pub fn parse(um_content: &str, mut config: Config) -> Self {
+        if um_content.is_empty() {
+            return Unimarkup {
+                doc: Document {
+                    blocks: vec![],
+                    config,
+                    ..Default::default()
+                },
+            };
+        }
+
```

Comment:
Is this necessary? Wouldn't the parser finish early returning document with empty vec when content is empty? If so maybe we could spare ourselves this part.

Reply:
not sure 😄 
I think in a previous version we had a panic parsing empty content.
If this is not the case anymore, I will remove it here.